projects/lights_control/src/can_setup.c

```diff
@@ -0,0 +1,91 @@
+#include "can_setup.h"
+#include "can.h"
+#include "can_hw.h"
+#include "can_msg_defs.h"
+
+#include "can_unpack_impl.h"
+#include "event_queue.h"
+#include "status.h"
+#include "structs.h"
+
+#define SOMEPORT 0
+#define SOMEPIN 0
+
+#define CASE_ACTION(action)            \
```

Comment:
I get what you're trying to do here, but this macro-magic is something that we generally try to avoid.

You can probably get rid of this by creating a guard function that checks if the event id is in the set of allowed events that the board is going to process, and then if it passes, then simply raise the event.